### PR TITLE
Migrate to govuk_tech_docs gem

### PR DESCRIPTION
**Pull request by himal-mandalia**
# Context

The DfE Technical Guidance is currently based on the [Jekyll static site generator](https://jekyllrb.com/) and published using [GitHub Pages](https://pages.github.com/).

# Changes proposed in this pull request

This pull request migrates the Technical Guidance to use the GDS [Tech Docs Template](https://github.com/alphagov/tech-docs-gem) which [The GDS Way](https://gds-way.cloudapps.digital/) and [other tech guidance](https://ministryofjustice.github.io/technical-guidance/) are based on.

The Tech Docs Template is based on the more powerful [Middleman static site generator](https://middlemanapp.com/) and allows for mixing of HTML, Markdown and ERB (embedded Ruby) in templates. For example, variables can be used:

```
---
title: Coding Principles
last_reviewed_on: 2020-12-21
review_in: 3 months
---

# <%= current_page.data.title %>
```
A search field with live search has also been added.

Screenshot 2020-12-22 at 12 06 02

This brings the DfE Technical Guidance in line with many others at GDS, MOJ and other departments. The Tech Docs Template is actively maintained by GDS.

## Changes in this pull request

* Update .gitignore
* Replace Jekyll with govuk_tech_docs gem in Gemfile
* Remove Makefile
* Add a layout under `source` folder
* Add javascript and stylesheet assets under `source` folder
* Restructure existing content files under `source` and `source/documentation` folders
* Rename and edit existing content files to `*.html.md.erb` to support Middleman features
* Add config in `config.rb` and `config/tech_docs.yml`
* Config includes redirects to support existing links, e.g. `/standards/naming-things` vs new `/documentation/naming-things.html` - both supported
* Update README
* Remove existing assets and content files used by Jekyll
* Add GitHub Action workflow to deploy to PaaS

# Guidance to review

Checkout the branch locally and follow the instructions in the README to set up and run the guidance. Review the running pages and check for any errors and inconsistencies.

### CODE REVIEW

**Comment by saliceti** [comment]
Tested locally successfully and reviewed content
Added a commit with small fixes for broken content

File: Makefile [hunk 1]
```diff
@@ -1,2 +0,0 @@
-serve:
```
Comment (saliceti) [inline Makefile:1]: It would be nice to keep the Makefile for middleman server and build

Comment (himal-mandalia) [reply Makefile:1]: Good point, feel free to add a Makefile back in. Also we need to discuss the best way to automate the build/publishing.

Comment (saliceti) [reply Makefile:1]: Added Makefile back in with `server` and `build` commands

**Comment by saliceti** [comment]
Serving with Github pages required a hack so we are changing the tech guidance to use the standard GDS tech docs gems. It’s also been moved to paas and served with an official education.gov.uk domain: https://technical-guidance.education.gov.uk/

**Pull request merged**